client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java

```diff
@@ -92,6 +95,26 @@ public static String appUniqueId(SparkContext context) {
         .getOrElse(context::applicationId);
   }
 
+  public static int celebornShuffleId(
```

Comment:
IMO it's better to move this method to `ShuffleClient` for potentially other engines. We can change the argument list to `appShuffleId, appShuffleIdentifier, isWriter`.

Reply:
`public abstract int getShuffleId(int appShuffleId, String appShuffleIdentifier, boolean isWriter);`
was already added to ShuffleClient 

I prefer to keep SparkUtils.celebornShuffleId() method as a place to check whether it needs to get shuffleId from LifecycleManager
```
  public static int celebornShuffleId(
      ShuffleClient client,
      CelebornShuffleHandle<?, ?, ?> handle,
      TaskContext context,
      Boolean isWriter) {
    if (handle.throwsFetchFailure()) {
      String appShuffleIdentifier = getAppShuffleIdentifier(handle.shuffleId(), context);
      return client.getShuffleId(handle.shuffleId(), appShuffleIdentifier, isWriter);
    } else {
      return handle.shuffleId();
    }
  }
```
otherwise, it would be hard to implement the check/fallback logic in place like constructor
```
  public SortBasedShuffleWriter(
      CelebornShuffleHandle<K, V, C> handle,
      TaskContext taskContext,
      CelebornConf conf,
      ShuffleClient client,
      ShuffleWriteMetricsReporter metrics,
      ExecutorService executorService,
      SendBufferPool sendBufferPool)
      throws IOException {
    this(
        SparkUtils.celebornShuffleId(client, handle, taskContext, true),
        handle.dependency(),
        handle.numMappers(),
        taskContext,
        conf,
        client,
        metrics,
        executorService,
        sendBufferPool);
  }
```

Reply:
We can add `shuffleId` as an argument to the constructor, as `HashBasedShuffleWriter` does. `SparkUtils.celebornShuffleId()` is a small wrapper for `ShuffleClient.getShuffleId`. I'm fine with this anyway.